src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -115,13 +117,28 @@ public boolean isSuccess() {
   }
 
   private void checkRetryable() {
-    Gson gson = new Gson();
-    JsonObject jo = gson.fromJson(this.responseBody, JsonObject.class);
-    if (jo.get(ServiceNowConstants.STATUS) != null &&
-      jo.get(ServiceNowConstants.STATUS).getAsString().equals(ServiceNowConstants.FAILURE) &&
-      jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString()
-        .contains(ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED)) {
+    if (httpStatus == HTTP_STATUS_TOO_MANY_REQUESTS) {
       isRetryable = true;
+      this.responseBody =
+              String.format(
+                      JSON_ERROR_RESPONSE_TEMPLATE,
+                      "Too many requests to ServiceNow API - decrease concurrent requests");
+    }
+    Gson gson = new Gson();
+    try {
+      JsonObject jo = gson.fromJson(this.responseBody, JsonObject.class);
+      if (jo.get(ServiceNowConstants.STATUS) != null
+              && jo.get(ServiceNowConstants.STATUS).getAsString().equals(ServiceNowConstants.FAILURE)
+              && jo.getAsJsonObject(ServiceNowConstants.ERROR)
+              .get(ServiceNowConstants.MESSAGE)
+              .getAsString()
+              .contains(ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED)) {
```

Comment:
Can you check if we require to handle null for the following in the responseBody json ?

1. Missing ERROR node
2. Missing MESSAGE node